rust/src/interpreter/run_result.rs

```diff
@@ -66,3 +66,14 @@ impl<'a> From<RunResult<'a>> for StepResult {
         )
     }
 }
+
+impl<'a> From<RunResult<'a>> for ExecutionResult {
```

Comment:
I was reviewing this code and I had to expand the file. 
```rust
impl<'a> From<StatusCode> for RunResult<'a> {
    fn from(status_code: StatusCode) -> Self {
        let step_status_code = match status_code {
            StatusCode::EVMC_SUCCESS => StepStatusCode::EVMC_STEP_RUNNING,
            StatusCode::EVMC_REVERT => StepStatusCode::EVMC_STEP_REVERTED,
            _ => StepStatusCode::EVMC_STEP_FAILED,
        };
        Self {
            step_status_code,
            status_code,
            revision: Revision::EVMC_FRONTIER,
            code_state: CodeState::new(&[], 0),
            gas_left: 0,
            gas_refund: 0,
            output: None,
            stack: Stack::new(Vec::new()),
            memory: Memory::new(Vec::new()),
            last_call_return_data: None,
        }
    }
}
```

This is fishy, we turn status code into EVMC_FRONTIER, but I don't see why we need revision in the result. Now we are casting `ExecutionResult` into `ExecutionResult`. do we even need that type?

Reply:
In an upcomming refactor PR I removed the RunResult completely but did not want to make this PR even larger. So maybe ignore this for now?

Reply:
looking forward.